### PR TITLE
TYP: added explanation of the join methods within df.align

### LINKED ISSUE #53092: DOC: pd.DataFrame.align() join parameter values are insufficiently explained
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.align.html

### Documentation problem

For the `join` parameter, only its values are mentioned. But what do the different values do? The current documentation provides no clue.

### Suggested fix for documentation

Please explain what the different values of the `join` parameter do.

### PR DESCRIPTION
**Pull request by Antony-evm**
- closes #53092 

Added type annotations of the join methods within df.align - ([here](https://github.com/Antony-evm/pandas-aevmorfop/blob/0.21.x/pandas/core/generic.py))

from  join : {{'outer', 'inner', 'left', 'right'}}, default 'outer'
           
To
 join : {{'outer', 'inner', 'left', 'right'}}, default 'outer'
 Type of alignment to be performed.

            left: use only keys from left frame, preserve key order.
            right: use only keys from right frame, preserve key order.
            outer: use union of keys from both frames, sort keys lexicographically.
            inner: use intersection of keys from both frames, preserve the order of the left keys.

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -9450,6 +9450,14 @@ def align(
         ----------
         other : DataFrame or Series
         join : {{'outer', 'inner', 'left', 'right'}}, default 'outer'
+            Type of alignment to be performed.
+
+            left: use only keys from left frame, preserve key order.
+            right: use only keys from right frame, preserve key order.
+            outer: use union of keys from both frames, sort keys lexicographically.
+            inner: use intersection of keys from both frames,
```
Comment (mroeschke) [inline pandas/core/generic.py:9458]:
```suggestion
            - left: use only keys from left frame, preserve key order.
            - right: use only keys from right frame, preserve key order.
            - outer: use union of keys from both frames, sort keys lexicographically.
            - inner: use intersection of keys from both frames,
```

**Comment by Antony-evm** [comment]
Im getting this error with the proposed changes and I fail to see how this could be caused by the documentation:
_error    libmamba Bad conversion of configurable 'remote_connect_timeout_secs' from source '~/.condarc' : yaml-cpp: error at line 14, column 30: bad conversion_

Any ideas? Is it a common issue?

**Comment by mroeschke** [comment]
Looks like theres a doc issue: 
```
2023-05-08T16:12:06.8973871Z /home/runner/work/pandas/pandas/pandas/core/generic.py:docstring of pandas.core.generic.NDFrame.align:13: WARNING: Bullet list ends without a blank line; unexpected unindent.
2023-05-08T16:12:06.8974978Z /home/runner/work/pandas/pandas/pandas/core/generic.py:docstring of pandas.core.generic.NDFrame.align:13: WARNING: Bullet list ends without a blank line; unexpected unindent.
```

Also no need to worry about that warning

**Comment by Antony-evm** [comment]
> Looks like theres a doc issue:
> 
> ```
> 2023-05-08T16:12:06.8973871Z /home/runner/work/pandas/pandas/pandas/core/generic.py:docstring of pandas.core.generic.NDFrame.align:13: WARNING: Bullet list ends without a blank line; unexpected unindent.
> 2023-05-08T16:12:06.8974978Z /home/runner/work/pandas/pandas/pandas/core/generic.py:docstring of pandas.core.generic.NDFrame.align:13: WARNING: Bullet list ends without a blank line; unexpected unindent.
> ```
> 
> Also no need to worry about that warning

Thank you for this - the last line of the list should have been indented apparently. Fixed it and it should be ready to close now. If you could point me to where to look in the logs next time, I'd appreciate it as I spent some time trying to find the error and couldn't find the above for the life of me.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @Antony-evm

**Pull request merged**